app/src/main/java/com/quran/labs/androidquran/ui/translation/TranslationView.java

```diff
@@ -68,6 +76,46 @@ public void onScrolled(RecyclerView recyclerView, int dx, int dy) {
     ayahToolBar.setVisibility(View.GONE);
     addView(ayahToolBar, LayoutParams.WRAP_CONTENT,
         context.getResources().getDimensionPixelSize(R.dimen.toolbar_total_height));
+
+    ImageButton copyButton = (ImageButton) ayahToolBar.findViewById(R.id.cab_copy_ayah_text_menu);
+    PopupMenu popup = new PopupMenu(context,copyButton);
+
+    copyButton.setOnClickListener(new View.OnClickListener() {
+      @Override
+
+      public void onClick(View v) {
+        PopupMenu popmenu;
+        popmenu = new PopupMenu(context, v);
+          int items = translations.length;
+          String[] titles = new String[items];
+        for (int i = 0; i < items; i++) {
+          popmenu.getMenu().add(Menu.NONE, i, 1, translations[i]);
```

Comment:
cache `popmenu.getMenu()` outside of the loop

Reply:
you mean like: "var objMenu=popupmenu.getMenu()"   and in for loop  just = "objMenu.add(.....)?"

Reply:
yes like that. but please fix indentation and combine 87 and 88 - i.e.
`final PopupMenu popupmenu = new PopupMenu(context, v);`